### PR TITLE
Fix a bug in reduction of element modulo ideal in exterior algebra

### LINKED ISSUE #37108: Bug in reduction of exterior algebra elements modulo ideal
### Steps To Reproduce

In Sage 10.2 and 10.2.rc4, reduction of elements of exterior algebra modulo an ideal does not give a reduced form.

```
sage: N = 4
sage: E = ExteriorAlgebra(QQ,binomial(N,2))
sage: e = E.gens()
sage: K = matroids.CompleteGraphic(N);
sage: C = K.circuits()
sage: ideal_gens = [sum([(-1)^j*E.prod(e[i] for i in list(c)[:j] + list(c)[j+1:]) for j in range(len(c))]) for c in C]
sage: OS = E.quo(ideal_gens)
sage: [OS(b).lift() for b in E.homogeneous_component_basis(4)]
[0,
 0,
 0,
 0,
 0,
 -e0*e1*e3*e4 + e0*e1*e3*e5,
 0,
 0,
 0,
 0,
 0,
 0,
 e0*e1*e3*e4 - e0*e1*e3*e5,
 e0*e1*e3*e4 - e0*e1*e3*e5,
 0]
sage: [OS(OS(b).lift()) for b in E.homogeneous_component_basis(4)]
[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
```

### Expected Behavior

There should be no nonzero degree-4 elements in the previous example.

### Actual Behavior

One needs to do the reduction step manually twice in order to reduce elements modulo the ideal. 

### Additional Information

A guess at the bug is that it could be caused by a loop terminating early during the Groebner basis reduction step.

### Environment

```markdown
- **OS**: MacOS Ventura 13.4.1
- **Sage Version**: 10.2, 10.2.rc4
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by trevorkarn**
This adds a loop to continue doing reduction modulo a Grobner basis for an element modulo an ideal in the exterior algebra. This guarantees a unique representative for each element, and that representative will be $0$ if the element is in the ideal. This fixes #37108.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### :hourglass: Dependencies

None.

### CODE REVIEW

File: src/sage/algebras/exterior_algebra_groebner.pyx [hunk 1]
```diff
@@ -477,13 +477,32 @@ cdef class GroebnerStrategy:
             0
             sage: I._groebner_strategy.reduce(E.zero())
             0
+
+        Check #37108 is fixed::
```
Comment (tscrim) [inline src/sage/algebras/exterior_algebra_groebner.pyx]:
```suggestion
        Check :issue:`37108` is fixed::
```

File: src/sage/algebras/exterior_algebra_groebner.pyx [hunk 2]
```diff
@@ -477,13 +477,32 @@ cdef class GroebnerStrategy:
...
+
+            sage: E = ExteriorAlgebra(QQ, 6)
+            sage: E.inject_variables(verbose=False)
+            sage: gens = [-e0*e1*e2 + e0*e1*e5 - e0*e2*e3 - e0*e3*e5 + e1*e2*e3 + e1*e3*e5,
+            ....:  e1*e2 - e1*e5 + e2*e5,
+            ....:  e0*e2 - e0*e4 + e2*e4,
+            ....:  e3*e4 - e3*e5 + e4*e5,
+            ....:  e0*e1 - e0*e3 + e1*e3]
```
Comment (tscrim) [inline src/sage/algebras/exterior_algebra_groebner.pyx]:
```suggestion
            sage: gens = [-e0*e1*e2 + e0*e1*e5 - e0*e2*e3 - e0*e3*e5 + e1*e2*e3 + e1*e3*e5,
            ....:         e1*e2 - e1*e5 + e2*e5, e0*e2 - e0*e4 + e2*e4,
            ....:         e3*e4 - e3*e5 + e4*e5, e0*e1 - e0*e3 + e1*e3]
```

**Review by tscrim — CHANGES_REQUESTED** [review]
Thank you. Some minor things.

Does this also result in a reduced GB for the example you gave on #37108?

**Comment by trevorkarn** [comment]
> Thank you. Some minor things.

Fixed - thanks!

> Does this also result in a reduced GB for the example you gave on #37108?

No it does not.

```
sage: N = 4
sage: E = ExteriorAlgebra(QQ,binomial(N,2))
sage: e = E.gens()
sage: K = matroids.CompleteGraphic(N);
sage: C = K.circuits()
sage: ideal_gens = [sum([(-1)^j*E.prod(e[i] for i in list(c)[:j] + list(c)[j+1:]) for j in range(len(c))]) for c in C]
sage: OS = E.quo(ideal_gens)
sage: I = OS.defining_ideal()
sage: I.groebner_basis()
(e0*e1 - e0*e3 + e1*e3,
 e0*e2 - e0*e4 + e2*e4,
 e1*e2 - e1*e5 + e2*e5,
 -e0*e1*e2 + e0*e1*e5 - e0*e2*e3 - e0*e3*e5 + e1*e2*e3 + e1*e3*e5,
 e3*e4 - e3*e5 + e4*e5)
sage: GS = I._groebner_strategy
sage: GS.groebner_basis
(e0*e1 - e0*e3 + e1*e3,
 e0*e2 - e0*e4 + e2*e4,
 e1*e2 - e1*e5 + e2*e5,
 -e0*e1*e2 + e0*e1*e5 - e0*e2*e3 - e0*e3*e5 + e1*e2*e3 + e1*e3*e5,
 e3*e4 - e3*e5 + e4*e5)
sage: GS.reduce_computed_gb()
sage: GS.groebner_basis
(e0*e1 - e0*e3 + e1*e3,
 e0*e2 - e0*e4 + e2*e4,
 e1*e2 - e1*e5 + e2*e5,
 e3*e4 - e3*e5 + e4*e5)
```

**Review by tscrim — APPROVED** [review]
This is definitely the more important bug to fix, so let's get this in.

However, we should still fix that the GB is not reduced when it is claiming it is.

**Comment by trevorkarn** [comment]
Sounds like a plan! Thanks!

**Pull request merged**